File: src/p_3_creational_patterns/p_9_builder/FiatOrderBuilder.java
package p_5_creational_patterns.p_9_builder;

public class FiatOrderBuilder extends OrderBuilder{
    @Override
    public void setBrand(String name) {
        getOrderedCar().setBrand(new Brand(name));
    }

    @Override
    public void setModel(String name) {
        getOrderedCar().setModel(new Model(name));
    }

    @Override
    public void setColor(String name) {
        getOrderedCar().setColor(name);
    }

    @Override
    public void setPower(int power) {
        getOrderedCar().setPower(power);
    }
}

File: Net/03_creational_patterns/05_builder_pattern/FiatOrderBuilder.cs
using System;
namespace _05_builder_pattern
{
    public class FiatOrderBuilder : OrderBuilder
    {
        public FiatOrderBuilder()
        {
        }

        public override void SetBrand(string name)
        {
            GetOrderedCar().SetBrand(new Brand(name));
        }

        public override void SetColor(string name)
        {
            GetOrderedCar().SetColor(name);
        }

        public override void SetModel(string name)
        {
            GetOrderedCar().SetModel(new Model(name));
        }

        public override void SetPower(int power)
        {
            GetOrderedCar().SetPower(power);
        }
    }
}

